besu/src/test/java/org/hyperledger/besu/PrivacyReorgTest.java

```diff
@@ -0,0 +1,558 @@
+
+    appendBlock(besuController, blockchain, protocolContext, secondBlock);
+
+    assertThat(privateStateStorage.getPrivacyGroupHeadBlockMap(secondBlock.getHash())).isNotEmpty();
+
+    assertThat(privateStateStorage.getPrivacyGroupHeadBlockMap(firstBlock.getHash()))
+        .contains(expected);
+  }
+
+  @Test
+  public void reorgToChainAtEqualHeight() {
+    // Setup an initial blockchain with one private transaction
+    final ProtocolContext<?> protocolContext = besuController.getProtocolContext();
+    final DefaultBlockchain blockchain = (DefaultBlockchain) protocolContext.getBlockchain();
+
+    final Block firstBlock =
+        gen.block(
+            new BlockDataGenerator.BlockOptions()
+                .setBlockNumber(1)
+                .setParentHash(blockchain.getGenesisBlock().getHash())
+                .addTransaction(generateMarker(getEnclaveKey(enclave.clientUrl())))
+                .hasOmmers(false)
+                .setReceiptsRoot(
+                    Hash.fromHexString(
+                        "0xc8267b3f9ed36df3ff8adb51a6d030716f23eeb50270e7fce8d9822ffa7f0461"))
+                .setGasUsed(23176)
+                .setLogsBloom(LogsBloomFilter.empty())
+                .setStateRoot(
+                    Hash.fromHexString(
+                        "0x16979b290f429e06d86a43584c7d8689d4292ade9a602e5c78e2867c6ebd904e")));
+
+    appendBlock(besuController, blockchain, protocolContext, firstBlock);
+
+    // Check that the private state root is not the empty state
+    assertPrivateStateRoot(
+        privateStateRootResolver,
+        blockchain,
+        "0x2121b68f1333e93bae8cd717a3ca68c9d7e7003f6b288c36dfc59b0f87be9590");
+
+    // Create parallel fork of length 1 which removes privacy marker transaction
+    final BlockDataGenerator.BlockOptions options =
+        new BlockDataGenerator.BlockOptions()
+            .setBlockNumber(firstBlock.getHeader().getNumber())
+            .setParentHash(blockchain.getGenesisBlock().getHash())
+            .hasTransactions(false)
+            .hasOmmers(false)
+            .setDifficulty(blockchain.getChainHeadHeader().getDifficulty().plus(10L))
+            .setReceiptsRoot(
+                Hash.fromHexString(
+                    "0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421"))
+            .setGasUsed(0)
+            .setLogsBloom(LogsBloomFilter.empty())
+            .setStateRoot(
+                Hash.fromHexString(
+                    "0x1bdf13f6d14c7322d6e695498aab258949e55574bef7eac366eb777f43d7dd2b"));
+    final Block forkBlock = gen.block(options);
+
+    appendBlock(besuController, blockchain, protocolContext, forkBlock);
+
+    // Check that the private state root is the empty state
+    assertPrivateStateRoot(
+        privateStateRootResolver,
+        blockchain,
+        "0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421");
```

Comment:
If `0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421` always represent the empty private state root, we could extract it to a constant to make the test easier to read.